src/menu/wallet_menu.c

```diff
@@ -120,23 +122,24 @@ static void wallet_menu_handler(engine_ctx_t *ctx,
  *
  * @param ctx The engine context* from which the flow is invoked
  * @param p0_evt The p0 event object which triggered the callback
- * @param data_ptr Data pointer here represents the wallet id of selected wallet
+ * @param data_ptr Data pointer here represents the copy of @ref Flash_Wallet
+ * object of selected wallet
  */
 static void ignore_p0_handler(engine_ctx_t *ctx,
                               p0_evt_t p0_evt,
                               const void *data_ptr);
 /*****************************************************************************
  * STATIC VARIABLES
  *****************************************************************************/
-static uint8_t wallet_id[WALLET_ID_SIZE] = {0};
+static Flash_Wallet flash_wallet = {0};
```

Comment:
Can you check if there is already a global variable of type `Flash_Wallet` present in the code? Aim should be to reduce duplicity..

Also, do you think we require the complete structure? Like pow_challenge` data? Or just the states/ wallet id/ name

Reply:
Picked the `Flash_Wallet` structure as it contains all important state variables. The function `get_flash_wallet_from_list_selection` returns the pointer to global variable of that Flash_Wallet object in flash_ram_instance.

We don't require the complete structure, the share and challenge are unnecessary for this purpose.

Reply:
In that case saving a copy of Flash_Wallet seems duplicate.. as we can fetch from existing functions everytime we require the info.. not making this comment blocking though..

Reply:
In that case, would it work to just keep the flash_wallet object pointer for that wallet?

Reply:
updated